### PR TITLE
Correct latent output reporting for AirflowNetwork Infiltration and Mixing

### LINKED ISSUE #7162: AirFlowNetwork output variables for Infiltration and Mixing Latent Gain/Loss are missing heat of vaporization
Issue overview
--------------
As noted by @XuanLuoLBL in https://github.com/NREL/EnergyPlus/pull/7040#discussion_r254387314 the heat of vaporization term appears to be missing from the calculations for output variables:
AFN Zone Infiltration Latent Heat Gain Rate and similar (Gain Energy, Loss Rate, and Loss Energy) and AFN Zone Mixing Latent Heat Gain Rate and similar.

These are calculated [here and following](https://github.com/NREL/EnergyPlus/blob/a2bd500d83ad361e2860eba8a03d4d1beae2b0d1/src/EnergyPlus/AirflowNetworkBalanceManager.cc#L8179-L8186).

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus v9.0.1-a2bd500d83ad361e2860eba8a03d4d1beae2b0d1 

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Correct results for latent output variables by multiplying Hfg for reports shown below. Previous reports showed kg/s, this update not reports Watts.

  Output:Variable,*,AFN Zone Infiltration Latent Heat Gain Rate,timestep;
  Output:Variable,*,AFN Zone Infiltration Latent Heat Gain Rate,timestep;
  Output:Variable,*,AFN Zone Infiltration Latent Heat Gain Energy,timestep;
  Output:Variable,*,AFN Zone Infiltration Latent Heat Loss Rate,timestep;
  Output:Variable,*,AFN Zone Infiltration Latent Heat Loss Energy,timestep;
  Output:Variable,*,AFN Zone Mixing Latent Heat Gain Rate,timestep;
  Output:Variable,*,AFN Zone Mixing Latent Heat Gain Energy,timestep;
  Output:Variable,*,AFN Zone Mixing Latent Heat Loss Rate,timestep;
  Output:Variable,*,AFN Zone Mixing Latent Heat Loss Energy,timestep;

The fixes #7162.

  
### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 1]
```diff
@@ -8286,6 +8287,7 @@ namespace AirflowNetworkBalanceManager {
                 if (ZN1 > 0 && ZN2 == 0) {
                     Tamb = Zone(ZN1).OutDryBulbTemp;
                     CpAir = PsyCpAirFnWTdb(OutHumRat, Tamb);
+                    hfg = Psychrometrics::PsyHfgAirFnWTdb(ZoneAirHumRat(ZN1), (MAT(ZN1) + Tamb) / 2.0);
```
Comment (rraustad) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:8298]: @Nigusse is this the correct Hfg funciton?

File: tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc [hunk 2]
```diff
@@ -2348,6 +2346,38 @@ TEST_F(EnergyPlusFixture, TestAFNPressureStat)
...
+    AirflowNetwork::AirflowNetworkLinkSimu(10).FLOW2 = 0.15;
+    AirflowNetwork::AirflowNetworkLinkSimu(13).FLOW2 = 0.1;
+
+    ReportAirflowNetwork();
+
+    EXPECT_NEAR(34.3673036, AirflowNetwork::AirflowNetworkReportData(1).MultiZoneInfiLatGainW, 0.0001);
+    EXPECT_NEAR(36.7133377, AirflowNetwork::AirflowNetworkReportData(2).MultiZoneMixLatGainW, 0.0001);
+    EXPECT_NEAR(89.3450925, AirflowNetwork::AirflowNetworkReportData(3).MultiZoneInfiLatLossW, 0.0001);
+
+    AirflowNetwork::AirflowNetworkExchangeData.deallocate();
```
Comment (rraustad) [inline tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc]: Array deallocation at end of unit tests is not standard. All deallocations should be performed by clear_state functions.

Comment (lgu1234) [reply tst/EnergyPlus/unit/AirflowNetworkBalanceManager.unit.cc]: @rraustad The array is deallocated in clear_state() of Elements.cpp. I am going to remove it after resolving conflict.

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc [hunk 3]
```diff
@@ -8642,23 +8647,11 @@ namespace AirflowNetworkBalanceManager {
...
-            AirflowNetworkZnRpt(i).MeanAirTemp = MAT(i);
-            AirflowNetworkZnRpt(i).OperativeTemp = 0.5 * (MAT(i) + MRT(i));
-        }
-
         for (i = 1; i <= NumOfZones; ++i) { // Start of zone loads report variable update loop ...
             Tamb = Zone(i).OutDryBulbTemp;
             CpAir = PsyCpAirFnWTdb(ZoneAirHumRatAvg(i), MAT(i));
             AirDensity = PsyRhoAirFnPbTdbW(OutBaroPress, MAT(i), ZoneAirHumRatAvg(i));
 
-            if (MAT(i) > Tamb) {
```
Comment (rraustad) [inline src/EnergyPlus/AirflowNetworkBalanceManager.cc:8655]: Why were these reports removed? MeanAirTemp, OperativeTemp, InfilHeatLoss and InfilHeatGain. Were they never reported before?

**Comment by rraustad** [comment]
This looks all good. A few minor questions. Why report variables were removed, and if hfg calculation needs to be updated to new function in a different branch. The hfg function does not necessarily need to be updated here, just that this specific calc MAY need updating in the near future.

File: src/EnergyPlus/AirflowNetworkBalanceManager.cc (hunk 3 above)
Comment (rraustad) [reply src/EnergyPlus/AirflowNetworkBalanceManager.cc:8655]: Are these report variables in the IO Ref and need to be removed?

Comment (lgu1234) [reply src/EnergyPlus/AirflowNetworkBalanceManager.cc:8655]: https://github.com/rraustad These variables do not have corresponding output variables. I am not sure "never", but these variables are not used anymore. Therefore, there is no need to revise IO Ref.

**Comment by rraustad** [comment]
Code changes reviewed and discussed. Comments have been addressed. These changes look good.

**Review by rraustad — APPROVED** [review]
Final review complete.

**Comment by rraustad** [comment]
Review of IO Ref for unused report variables as discussed herein. None found.

**Pull request merged**